packages/meditrak-server/src/routes/postChanges.js

```diff
@@ -92,12 +112,41 @@ async function validateSurveyResponseObject(models, surveyResponseObject) {
   }
 }
 
+async function translateSurveyResponseObject(models, surveyResponseObject) {
+  if (!surveyResponseObject) {
+    throw new ValidationError('Payload must contain survey_response_object');
+  }
+
+  const surveyResponseTranslators = constructSurveyResponseTranslators(models);
+  await translateObjectFields(surveyResponseObject, surveyResponseTranslators);
+
+  const { answers } = surveyResponseObject;
+  if (Array.isArray(answers)) {
+    const answerTranslators = constructAnswerTranslators(models);
+    for (let i = 0; i < answers.length; i++) {
+      await translateObjectFields(answers[i], answerTranslators);
```

Comment:
Could be quite a bit more efficient to fetch the question code to id mapping in one db query, and then use that mapping object during each translation

Reply:
While you're totally right about this, I'm thinking of maybe leaving it as is for now? Doing the fetch upfront would be a bit of a tricky code refactor, and given that this code will probably be totally refactored soon enough I'm okay with leaving it a little inefficient?

Reply:
Yep :+1: